### DISCUSSION TITLE
I want to use all ESP

### DISCUSSION THREAD

**Discussion by zhaoketong** [anymail/django-anymail#259] [Q&A]
If I want to use all ESP, how should I set it in setting.py and how should I choose which one to use

**Comment by medmunds** [comment]
See [*Mixing email backends*](https://anymail.readthedocs.io/en/stable/tips/multiple_backends/) in the Anymail docs for examples of sending different messages through different email providers, by supplying a `connection` argument to `send_mail()` or in an `EmailMessage()` constructor .

In your settings.py, you'll need to pick one ESP to be your default EMAIL_BACKEND. And then provide the required settings for each ESP in the `ANYMAIL` settings dict:

```python
# settings.py
EMAIL_BACKEND = "anymail.backends.mailgun.EmailBackend"  # your default choice
ANYMAIL = {
    "MAILGUN_API_KEY": "12345",
    "SENDGRID_API_KEY": "67890",
    # etc.
}
```

One caution: using multiple ESPs at the same time, for the same recipient addresses, can cause deliverability issues. If you're not very careful about how you mix ESPs, it increases the chance your messages will end up in spam folders. (And a high spam rate will quickly lead to ESPs banning you.) You'll need a way to synchronize block/bounce lists between ESPs, or to consistently partition email recipients among ESPs. There are some cases where multiple ESPs make sense, but it's a good idea to discuss your plans and needs with each of the ESPs involved in advance.

**Reply by zhaoketong** [reply]
I think I get it. Thank you
